Fix quiz feature

papes1ns opened the pull request:
# Quiz Behavior Spec (from #755)

This spec outlines decisions about quiz creation, editing, and instructor experience.

## Context

- Instructors deploy quizzes to assess learner understanding.
- Case authors create suggested quizzes before publishing.
- Instructors can assign suggested quizzes or create custom ones (with optional base questions + additions).

## Spec

- [ ] **Disallow empty quizzes**
  - Require a title and at least one question
  - Enforce in both UI and model

- [ ] **Distinguish suggested vs. custom quizzes in editor**
  - Visually differentiate instructor-created and system-generated (empty) quizzes

- [ ] **Restrict deletion of suggested quizzes**
  - No global or per-deployment deletion
  - Use visual indicators instead

- [ ] **Copy on edit for suggested quizzes**
  - Modifying a suggested quiz creates a private, editable copy

- [ ] **Improve instructor quiz preview**
  - Preferred: allow instructors to interact like learners
  - Fallback: turn off submit and add “preview” note

## Status

Item 1 addresses a system fix and is included in this pull request. The remaining items represent a working specification based on the discussions in #755.

waisaed commented:
Great improvements!

The only issue I'm seeing is that validation is firing on custom quizzes (through the deployment editor) that do have >0 questions. 

I'm also going to tweak the '_Base Quiz must have at least one question (either custom or inherited)_' validation language, because I'm not sure users will understand it.

papes1ns commented:
Added fix for the Toolbar component for the deployment editor. It now persists the toggles in the correct state as in the database.

<img width="469" alt="Screenshot 2025-05-23 at 9 52 13 PM" src="https://github.com/user-attachments/assets/8a36a89e-4495-438e-af79-a66e0a53fb3e" />


Also, added fix for the sentry serialization error
```
object.js:106 Uncaught (in promise) TypeError: Illegal invocation
    at JSON.stringify (<anonymous>)
    at u (object.js:106:17)
    at e.<anonymous> (beacon.ts:14:18)
    at bundle.min.js:2:1764
    at Object.next (bundle.min.js:2:1869)
    at bundle.min.js:2:807
    at new Promise (<anonymous>)
    at o (bundle.min.js:2:584)
    at e.captureEvent (beacon.ts:13:29)
    at e.<anonymous> (backend.ts:157:27)
    at bundle.min.js:2:1764
    at Object.next (bundle.min.js:2:1869)
    at bundle.min.js:2:807
    at new Promise (<anonymous>)
    at o (bundle.min.js:2:584)
    at e.sendEvent (backend.ts:134:26)
    at e.<anonymous> (baseclient.js:202:77)
    at baseclient.js:43:23
    at Object.next (baseclient.js:24:53)
    at baseclient.js:18:71
    at new Promise (<anonymous>)
    at r (baseclient.js:14:12)
    at baseclient.js:201:107
    at e.<anonymous> (baseclient.js:379:46)
    at baseclient.js:43:23
    at Object.next (baseclient.js:24:53)
    at a (baseclient.js:15:58)
```
